### PR TITLE
Add initial runtime/sbf/syscalls.md

### PR DESCRIPTION
**Pull request by riptl**
**Summary**

Relates to #11 

- Adds Python script for generating syscall table
- Adds syscall table
- Documents the following syscalls
  - abort
  - sol_panic_
  - sol_log_
  - sol_log_64_
- Documents syscall memory accesses subroutines

**Reviewers**

- *Required:* Solana Labs VM team: @jackcmay @Lichtso @alessandrod

**Organization**

- *Section:* Runtime
- *Status:* Standard, live on mainnet

### CODE REVIEW

File: runtime/sbf/syscalls.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+| `3770fb22` | `sol_memset_`                           |
+| `a22b9c85` | `sol_invoke_signed_c`                   |
+| `d7449092` | `sol_invoke_signed_rust`                |
+| `83f00e8f` | `sol_alloc_free_`                       |
+| `a226d3eb` | `sol_set_return_data`                   |
+| `5d2245e4` | `sol_get_return_data`                   |
+| `7317b434` | `sol_log_data`                          |
+| `adb8efc8` | `sol_get_processed_sibling_instruction` |
+| `85532d94` | `sol_get_stack_height`                  |
+
```
Comment (Lichtso) [inline runtime/sbf/syscalls.md:57]: In ABIv2 some syscalls will be removed / replaced and some will be added:

https://github.com/solana-labs/solana/blob/c68fb5f68db5677809fad750487b92252ea763d8/programs/bpf_loader/src/syscalls/mod.rs#L248

File: runtime/sbf/syscalls.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+| `3770fb22` | `sol_memset_`                           | ☑    | ☑    |
+| `a22b9c85` | `sol_invoke_signed_c`                   | ☑    | ☐    |
+| `d7449092` | `sol_invoke_signed_rust`                | ☑    | ☐    |
+| `83f00e8f` | `sol_alloc_free_`                       | ☑    | ☐    |
+| `a226d3eb` | `sol_set_return_data`                   | ☑    | ☐    |
+| `5d2245e4` | `sol_get_return_data`                   | ☑    | ☐    |
+| `7317b434` | `sol_log_data`                          | ☑    | ☑    |
+| `adb8efc8` | `sol_get_processed_sibling_instruction` | ☑    | ☐    |
+| `85532d94` | `sol_get_stack_height`                  | ☑    | ☐    |
+| `????????` | `sol_set_account_attributes`            | ☐    | ☑    |
```
Comment (ripatel-jump) [inline runtime/sbf/syscalls.md:57]: TODO hash

File: runtime/sbf/syscalls.md (hunk 1 above)
Comment (ripatel-jump) [reply runtime/sbf/syscalls.md:58]: Thank you for the pointer, fixed.

File: runtime/sbf/syscalls.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+| `dd1c41a6` | `sol_curve_group_op`                    | ☑    | ☑    |
+| `d56b5fe9` | `sol_get_clock_sysvar`                  | ☑    | ☑    |
+| `23a29a61` | `sol_get_epoch_schedule_sysvar`         | ☑    | ☑    |
+| `3b97b73c` | `sol_get_fees_sysvar`                   | ☑    | ☑    |
+| `bf7188f6` | `sol_get_rent_sysvar`                   | ☑    | ☑    |
+| `717cc4a3` | `sol_memcpy_`                           | ☑    | ☑    |
+| `434371f8` | `sol_memmove_`                          | ☑    | ☑    |
+| `5fdcde31` | `sol_memcmp_`                           | ☑    | ☑    |
+| `3770fb22` | `sol_memset_`                           | ☑    | ☑    |
+| `a22b9c85` | `sol_invoke_signed_c`                   | ☑    | ☐    |
```
Comment (ripatel-jump) [inline runtime/sbf/syscalls.md:49]: Empty ballot check box renders weird in markdown

File: runtime/sbf/syscalls.md (hunk 2 above)
Comment (ripatel-jump) [reply runtime/sbf/syscalls.md]: Done, added a generator script now.

File: runtime/sbf/syscalls.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+Syscalls are VM hardcoded subroutines extending the functionality of the virtual machine.
+They generally belong to one of two categories:
+
+- *Runtime syscalls* provide access to external facilities (e.g. logging)
+- *Convenience syscalls* provide efficient replacements for functionality that could be implemented without syscalls (e.g. SHA-256 hashing)
+
+## Usage
+
+A syscall is invoked using a [`call` instruction](./isa.md#call) with `dst=0`, `src=0`, `off=0`, and `imm` set to the hash in reverse byte order.
```
Comment (Lichtso) [inline runtime/sbf/syscalls.md:11]: The way syscalls and normal function calls are differentiated was changed in SBFv2, see `config.static_syscalls`: https://github.com/solana-labs/rbpf/blob/e58129a441f306d024e8e7397428c0fb96df2224/src/interpreter.rs#L446

**Pull request closed without merging**